src/ai/backend/agent/stats.py

```diff
@@ -449,28 +463,34 @@ async def collect_node_stat(self) -> None:
                 stage="before_observe",
                 upper_layer="collect_node_stat",
             )
-            for result in results:
-                if isinstance(result, BaseException):
-                    log.error("collect_node_stat(): gather_node_measures() error", exc_info=result)
+            for res in results:
+                if isinstance(res, BaseException):
+                    log.error("collect_node_stat(): gather_node_measures() error", exc_info=res)
                     continue
+                slot_names, result = res
                 for node_measure in result:
                     metric_key = node_measure.key
                     # update node metric
+                    slot_scale_factor = self._extract_slot_measurement_scale_factor(
+                        resource_scaling_factors, slot_names, metric_key
+                    )
+                    per_node = node_measure.per_node.apply_scale_factor(slot_scale_factor)
```

Comment:
Isn't it okay to call `_extract_slot_measurement_scale_factor` outside the for loop?